### ISSUE TITLE
fix: seed model init from a local generator instead of torch.manual_seed

### ISSUE THREAD

**Issue by g8rdier** [g8rdier/hqnn-forge#175]
`HybridClassifierEstimator.fit` seeds weight initialisation by calling `torch.manual_seed(random_state)` (`hqnn_forge/sklearn.py`), which reseeds the **global** torch RNG for the whole process. A caller who does

```python
torch.manual_seed(123)
...
clf.fit(X, y)          # random_state=0
other = torch.randn(3) # no longer the stream the caller set up
```

silently gets a different random stream after the fit than before it. Inside `GridSearchCV` or a `cross_val_score` loop, every fit reseeds the process, so anything else drawing from the global RNG in the same process is affected too.

Batch order is already handled correctly: `fit` passes an explicit `torch.Generator` to `train_model`. Only weight initialisation needs the global seed, because nothing downstream takes a generator:

- `HybridBinaryClassifier` / `ParallelHybridClassifier` do not accept a `generator` or `seed` argument.
- Neither do the init strategies they call.

## Proposed

Thread an optional `torch.Generator` through the model constructors into the init strategies, and have the estimator pass one built from `random_state` instead of touching the global RNG. `torch.nn.init.*` and `torch.randn`-style draws take `generator=`, so the plumbing is mechanical; the decision to make is the constructor API (`generator: torch.Generator | None` vs. a plain `seed: int | None`).

Until then the behaviour is documented on the estimator's `random_state` parameter.

Found during the review of #130; deliberately deferred so it would not hold up that merge, since the fix is a library-wide constructor change rather than estimator work.
